src/main/java/hudson/tasks/Mailer.java

```diff
@@ -395,6 +422,7 @@ protected PasswordAuthentication getPasswordAuthentication() {
         @Override
         public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
 
+            // TODO try-with-resources for BulkChange instance?
```

Comment:
Same here, TODO here seems like cruft. My (non maintainer) vote is remove the TODO or fix the issue

Reply:
Didn't know if I should cover just one change in the plugin or use the fix-as-much-as-I-can approach. That's why I used TODOs